### ISSUE TITLE
Conflict when using an optional array with minItems set

### ISSUE THREAD

**Issue by ThomasAribart** [rjsf-team/react-jsonschema-form#3363]
### Prerequisites

- [X] I have searched the existing issues
- [X] I understand that providing a [SSCCE](http://sscce.org/) example is tremendously useful to the maintainers.
- [X] I have read the [documentation](https://react-jsonschema-form.readthedocs.io/)
- [ ] Ideally, I'm providing a [sample JSFiddle](https://jsfiddle.net/n1k0/f2y3fq7L/6/) or a [shared playground link](https://rjsf-team.github.io/react-jsonschema-form/) demonstrating the issue.

### What theme are you using?

material-ui

### Version

5.x

### Current Behavior

Hi and thanks for this great lib 🙌

When setting an optional property in an `"object"`, which is of type `"array"` with a `"minItem"` set to 1 or more, it's not possible to remove the property from the said object.

By default, the array property widget appears with items that can be empty, which can cause bugs.

### Expected Behavior

I don't have any design proposal, I know it's tricky to represent, but the default items should only appear when deciding that the property is not undefined somehow.

### Steps To Reproduce

```
const mySchema = {
  type: "object,
  properties: {
    optionalArray: {
      type: "array",
      items: { type: "string" },
      minItems: 2
    }
  }
}
```

### Environment

```markdown
- OS: Any
- Node: Any
- npm: Any
```

**Comment by heath-freenome** [comment]
Here's a [playground](https://rjsf-team.github.io/react-jsonschema-form/#eyJmb3JtRGF0YSI6eyJvcHRpb25hbEFycmF5IjpbbnVsbCxudWxsXX0sInNjaGVtYSI6eyJ0eXBlIjoib2JqZWN0IiwicHJvcGVydGllcyI6eyJvcHRpb25hbEFycmF5Ijp7InR5cGUiOiJhcnJheSIsIml0ZW1zIjp7InR5cGUiOiJzdHJpbmcifSwibWluSXRlbXMiOjJ9fX0sInVpU2NoZW1hIjp7fSwidGhlbWUiOiJkZWZhdWx0IiwibGl2ZVNldHRpbmdzIjp7InNob3dFcnJvckxpc3QiOiJ0b3AiLCJ2YWxpZGF0ZSI6ZmFsc2UsImRpc2FibGUiOmZhbHNlLCJyZWFkb25seSI6ZmFsc2UsIm9taXRFeHRyYURhdGEiOmZhbHNlLCJsaXZlT21pdCI6ZmFsc2V9fQ==) for it

**Comment by heath-freenome** [comment]
@ThomasAribart Good bug. The code is not taking into account the required-ness of the property. This seems like an easy fix in the `@rjsf/utils` `getDefaultFormState()` function's helper `computeDefaults()`. Are you willing to fix this for us? If so, I can point you in the right direction.

**Comment by kuosandys** [comment]
Hi there 👋 we’re also encountering the same bug in our project. Is there currently work in progress to fix it? If not, could we offer our help with fixing it?

**Comment by RPiAwesomeness** [comment]
Same here. Would be willing to help out to get this fixed.

**Comment by heath-freenome** [comment]
Yes please! Any help fixing issues is highly encouraged.

**Comment by RPiAwesomeness** [comment]
@heath-freenome Question for you. I'm working on this and ran into a situation where I think one of the existing tests conflicts with this bug. Here's the console output:

![image](https://user-images.githubusercontent.com/11481499/225015547-cf315c73-cef8-4de2-9a85-a29bec68a41f.png)

I think the consensus here is that if the field isn't required and has `minItems` set, that no items will be automatically added. Correct? In that case, I would argue that if the field isn't required and no items have been added by the user, then the field shouldn't be included in the output at all, thus making this test invalid. Having an empty array included in the data seems to contradict the `minItems` requirement.

Are you okay with me removing this test or is there a part of the JSON Schema spec that I'm missing that specifies this behavior?

**Comment by RPiAwesomeness** [comment]
Another one that's similar:

![image](https://user-images.githubusercontent.com/11481499/225034252-b4de4189-3f1b-4811-be83-1f7c93cd1380.png)

Seems odd that we're auto-filling default values when it's not marked required.

**Comment by heath-freenome** [comment]
@RPiAwesomeness These are great questions and merit a verbal conversation with the rest of the current maintainers (all 2 of us). Are you able to come to the weekly meeting that we have every Friday at 11am Eastern time? See this [post](https://github.com/rjsf-team/react-jsonschema-form/issues/2417) for the deets

**Comment by RPiAwesomeness** [comment]
Potentially. I'll double-check my schedule but that should work for me. Anything in particular that I should prep to help the discussion?

**Comment by heath-freenome** [comment]
> Potentially. I'll double-check my schedule but that should work for me. Anything in particular that I should prep to help the discussion?

Not really, if anything there might be a new minor feature that will need to be added to support the work you are doing in a backwards-compatible way. We can talk about it Friday and give you a bit more background than is here in this bug.
